samples/performance/async_compute/async_compute.cpp

```diff
@@ -38,6 +38,15 @@ AsyncComputeSample::AsyncComputeSample()
 	config.insert<vkb::BoolSetting>(1, double_buffer_hdr_frames, true);
 }
 
+void AsyncComputeSample::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+#ifdef VK_ENABLE_BETA_EXTENSIONS
+	// We need to enable the mutableComparisonSamplers feature of the VK_KHR_portability_subset extension
```

Comment:
Can you elaborate on this? Why do we need to enable this in this case?

Reply:
The portability feature `mutableComparisonSamplers` needs to be enabled for certain samples when running on MoltenVK on macOS/iOS, otherwise you get the following validation error...

_[error] 256275552 - VUID-VkDescriptorImageInfo-mutableComparisonSamplers-04450: Validation Error: [ VUID-VkDescriptorImageInfo-mutableComparisonSamplers-04450 ] | MessageID = 0xf467460 | vkUpdateDescriptorSets(): pDescriptorWrites[4] (portability error): sampler comparison not available. The Vulkan spec states: If the VK_KHR_portability_subset extension is enabled, and VkPhysicalDevicePortabilitySubsetFeaturesKHR::mutableComparisonSamplers is VK_FALSE, then sampler must have been created with VkSamplerCreateInfo::compareEnable set to VK_FALSE (https://vulkan.lunarg.com/doc/view/1.3.283.0/mac/1.3-extensions/vkspec.html#VUID-VkDescriptorImageInfo-mutableComparisonSamplers-04450)_

 The `VK_ENABLE_BETA_EXTENSIONS` define is currently synonymous with the `VK_KHR_portability_subset` extension (note this is not the same as `VK_KHR_portability_enumeration`), and as of this pull request is enabled only for Apple platforms.  If I am missing something and this is not correct, please let me know.

Reply:
If this is a MoltenVK-specific thing, it should be guareded by a check that says so. Using some defined that currently is synonymous to some other that currently is only enabled on that platform is probably not future-safe.

Reply:
Based on my understanding, I believe this is a Vulkan _portability_ thing and not a MoltenVK-specific thing.  MoltenVK just happens to be a leading example of portability.  Looking at the validation error message it states:

_If the VK_KHR_portability_subset extension is enabled, and VkPhysicalDevicePortabilitySubsetFeaturesKHR::mutableComparisonSamplers is VK_FALSE, then sampler must have been created with VkSamplerCreateInfo::compareEnable set to VK_FALSE_ 

The two examples (_async_compute_ and _multithreading_render_passes_) where I have enabled `mutableComparisonSamplers` create samplers with `compareEnable` set to VK_TRUE.  Thus the spec requires this feature to be enabled for _any_ portability implementation.

The current guard that enables the `VK_KHR_portability_subset` extension and its associated features is `VK_ENABLE_BETA_EXTENSIONS`.  This is already used in other places in the code.  For instance, look at the existing _portability_ sample, and also my addition to `vulkan_sample.h` which enables the `VK_KHR_portability_subset` extension which is also required by the spec for portability implementations.  I was just trying to stay consistent with the approach used by the existing framework and samples.

Note that `VKB_ENABLE_PORTABILITY` has a different purpose in the current framework, and that is to enable the `VK_KHR_portability_enumeration` extension, which is distinct from `VK_KHR_portability_subset`.